ironfish/src/wallet/validator.ts

```diff
@@ -76,6 +83,12 @@ export function validateAccount(toImport: Partial<AccountValue>): void {
   if (toImport.spendingKey && !isValidSpendingKey(toImport.spendingKey)) {
     throw new Error(`Provided spending key ${toImport.spendingKey} is invalid`)
   }
+
+  if (isValidIVKAndPublicAddressPair(toImport.incomingViewKey, toImport.publicAddress)) {
+    throw new Error(
+      `Public address ${toImport.publicAddress} does not match public address generated from incoming view key ${toImport.incomingViewKey}`,
```

Comment:
It might be nice to also log the address generated from the incoming view key -- might give someone a clue as to why they don't match.

Reply:
Done!